Console tips for VirtualBox FreeBSD guests (#162)
freebsd/freebsd-doc: documentation/content/en/books/handbook/virtualization/_index.adoc

# mount_vboxvfs -w myshare /mnt
....

> If you want to change the resolution of the console but `vidcontrol -i mode` returns an empty list, then you'll need to switch from the "vt" console driver to the "sc" console driver as explained in man:vidcontrol[1].

Reviewer: ```suggestion
If you want to change the resolution of the console, but `vidcontrol -i mode` returns an empty list: you may use the legacy man:syscons[4] alternative to the man:vt[4] console driver.
```

> Add the following line to [.filename]#/boot/loader.conf#:

Reviewer: ```suggestion
For syscons (sc), add the following line to [.filename]#/boot/loader.conf#:
```

[.programlisting]
....
kern.vty=sc
....

> Then, after rebooting, you'll be able to list available console video modes and select one with `vidcontrol` (be sure to adjust the video memory setting for this VM in VirtualBox parameters to a sufficient value). For example:

Reviewer: ```suggestion
Shut down the guest, then review its video memory setting.
Allow enough memory for the intended resolution.

After the guest boots, man:vidcontrol[1] will allow you to specify a listed mode.
For example:
```

[source,shell]
....
# vidcontrol -i mode
    mode#     flags   type    size       font      window      linear buffer
------------------------------------------------------------------------------
...
325 (0x145) 0x0000000f G 1280x1024x32 D  8x16  0xa0000 64k 64k 0xe0000000 5120k
...
# vidcontrol MODE_325
....

> If this new video mode is acceptable, it can be permanently set on boot for all virtual consoles by adding it to [.filename]#/etc/rc.conf#:

Reviewer: ```suggestion
If the new mode is acceptable, it can be set for all virtual consoles:
```

> [.programlisting]
> ....
> allscreens_flags="MODE_325"

Reviewer: ```suggestion
[source,shell]
....
# sysrc allscreens_flags+="MODE_325"
```

For the configuration to take effect, either `reboot -r` or restart the OS.

....

> If you are using a foreign keyboard, you also need to add appropriate keymap and font lines to [.filename]#/etc/rc.conf#.

Reviewer: ```suggestion
If you use a foreign keyboard, you should also add appropriate keymap and font lines to [.filename]#/etc/rc.conf#.
See man:rc.conf[5].
```

> Beware, the keymap files are different from "vt" to "sc" console types! Check [.filename]#/usr/share/syscons/keymaps# for appropriate values...

Reviewer: ```suggestion
Note, keymap files for syscons differ from those for vt.
For appropriate values, see files in [.filename]#/usr/share/syscons/keymaps#.
```

> For example, in order to set a French keyboard do:

Reviewer: ```suggestion
For example, to set a French keyboard:
```

[.programlisting]
....
keymap="fr.iso.acc"

> font8x16="iso-thin-8x16"

Reviewer: I assume that this format suits syscons (not vt).
